### PR TITLE
fix: Do not skip deleted files in regexp filter

### PR DESCRIPTION
**Pull request by smacker**
Ref: https://github.com/src-d/lookout/issues/445

Signed-off-by: Maxim Sukharev <max@smacker.ru>

### CODE REVIEW

**Comment by carlosms** [comment]
What if I request `IncludePattern = vendor*` but the file `cmd/main.go` gets deleted? Will it be included in the response with the change in this PR?

**Comment by smacker** [comment]
Good point.
I also see that in Vendor & Language filters we apply them on `Base` if `Head == nil`.

I've updated the PR implementing the same logic to be consistent and it actually makes sense at least to me.

File: service/git/scanner.go [hunk 1]
```diff
@@ -217,6 +217,10 @@ func NewChangeFilterScanner(scanner lookout.ChangeScanner, include, exclude stri
 	}
 
 	fn := func(ch *lookout.Change) (bool, error) {
+		if ch.Head == nil {
```
Comment (se7entyse7en) [inline service/git/scanner.go]: Can you please explain when this happen?

Comment (smacker) [reply service/git/scanner.go]: When you delete a file in git.

**Comment by carlosms** [comment]
LGTM. Should we have tests to check the filters apply to base when head is deleted?

**Pull request merged**